### PR TITLE
Add additional description for complex commands

### PR DESCRIPTION
**Pull request by johnallen3d**
Commands like `console` and `setup` support overrides and hooks. In order to make that more clear this commit adds a `.long_description` field to `./config/commands.json` and exposes that in the command specific help (ie. `nib console --help`).

After this change help for `console` and `setup` render as such:

```
❯ nibdev console --help
Usage: nib console [SERVICE]

Start a REPL session for the given service

This command will try to detect the applications environment (ie. rails console or bundle console)
and start the most appropriate REPL possible. It's possible to directly override this behaviour by
supplying an executable file at $pwd/bin/console.

Options:
  -e, --env   Environment name (default: development)
  -h          Print usage

Services:
  - app
```

```
❯ nibdev setup --help
Usage: nib setup [SERVICE]

Runs application specific setup for the given service

By default the setup command will execute 'bundle install && rake db:create db:migrate'.

This behavior can be overriden by providing an executable file in the project at $pwd/bin/setup.
Additionally the setup process can be augmented by providing either $pwd/bin/setup.before or
$pwd/bin/setup.after. This allows for extending the default behavior without having to redefine it.

Services:
  - app
```

### CODE REVIEW

File: bin/_tty [hunk 1]
```diff
@@ -4,19 +4,6 @@
...
-$(print_command_instructions)
-
-Options:
-  -e, --env   Environment name (default: development)
-  -h          Print usage
-
-Services:
-$(print_services)"
-}
-
```
Comment (johnallen3d) [inline bin/_tty]: This change is not strictly necessary for adding long description support but this function is help related and completely wrong. Can move to a separate commit if deemed necessary.

**Comment by johnallen3d** [comment]
@blankenshipz 👀

**Comment by blankenshipz** [comment]
I think that the `update_docs.sh` script and the corresponding `docs/commands.md` need to be updated, other then that it looks great!

**Comment by johnallen3d** [comment]
Can't say that I disagree @blankenshipz, I tried for a while to find a format that worked but didn't have much luck. Have any thoughts?

**Comment by johnallen3d** [comment]
Also, it occurs to me now that we were talking about adding a note in `nib help` next to commands w/ long description to see `nib [command] --help` for additional details. I'll take a look at adding that.

**Comment by johnallen3d** [comment]
Added the additional detail to `nib help` in this commit: 77fcbc7

**Comment by johnallen3d** [comment]
Maybe `docs/commands.md` should be in a format more [along these lines](https://docs.docker.com/engine/reference/commandline/commit/) vs. a table?

**Comment by blankenshipz** [comment]
@johnallen3d :+1: for that idea. I also think that a table with three columns is fine (command, description, additional info)

**Comment by johnallen3d** [comment]
The trouble  with three columns is that the long descriptions generally have several linebreaks. The table gets quite large and unbalanced in those cases.

**Comment by blankenshipz** [comment]
Yeah, I think you're on the 💰 with your idea to mimic the docker docs.

**Comment by johnallen3d** [comment]
Ok, the latest commit (1ed6056) produces something like this.

```md
# Supported Commands

The following commands are available:

### `bundle`

Run bundle for the given service

### `codeclimate`

Run codeclimate againt the current working directory

### `console`

Start a REPL session for the given service

This command will try to detect the applications environment (ie. rails console or bundle console)
and start the most appropriate REPL possible. It's possible to directly override this behaviour by
supplying an executable file at $pwd/bin/console.

### `debug`

Connect to a running byebug server for a given service
```

Which looks like this:

commands_md

It's not ideal yet (would like to call out to the help function for each command) but I think it's a step in the right direction. What say you @blankenshipz?

**Comment by blankenshipz** [comment]
:shipit:
